### PR TITLE
[FIX] account,snailmail_account: fix methods depends on env

### PR DESCRIPTION
**Pull request by clbr-odoo**
Some methods were using the `self.env` to decide users/partners while those methods can also be called from Cron, thus the user would end up to be OdooBot instead of the partner that initiated the async sending.

### CODE REVIEW

File: addons/snailmail_account/wizard/account_move_send.py [hunk 1]
```diff
@@ -77,16 +76,17 @@ def _hook_if_success(self, moves_data, from_cron=False, allow_fallback_pdf=False
...
-                for move in moves
+        to_send = {
+            move: move_data
+            for move, move_data in moves_data.items()
+            if move_data.get('send_by_post') and move.invoice_pdf_report_id
+        }
+        if to_send:
+            self.env['snailmail.letter'].create([
+                {
+                    'user_id': self.env['res.partner'].browse(move_data['sp_partner_id']).user_id.id,
```
Comment (smetl) [inline addons/snailmail_account/wizard/account_move_send.py:87]: This could be a performance issue. Backup the user id as well instead?

Comment (clbr-odoo) [reply addons/snailmail_account/wizard/account_move_send.py:87]: Sure, but we're in stable, we might have pending invoices to send that don't have the newly `sp_user_id` set, what should be the fallback value ? :thinking:

Comment (smetl) [reply addons/snailmail_account/wizard/account_move_send.py]: @clbr-odoo odoobot I guess.... we have no choice anyway :sweat_smile:

**Review by smetl — APPROVED** [review]
@robodoo r+

**Comment by Xavier-Do** [comment]
@robodoo retry

**Pull request closed without merging**